src/rpc.ts

```diff
@@ -128,18 +128,22 @@ export class RPC extends EventEmitter {
         return;
       }
 
-      // tslint:disable-next-line
-      Promise.resolve(handler(data.params)).then(result => {
-        const packet: IRPCReply<any> = {
-          type: 'reply',
-          serviceID: RPC.serviceID,
-          id: data.id,
-          result,
-        };
+      const packet: IRPCReply<any> = {
+        type: 'reply',
+        serviceID: RPC.serviceID,
+        id: data.id,
+        result: null,
+      };
 
-        this.emit('sendReply', packet);
-        this.post(packet);
-      });
+      Promise.resolve()
+        .then(() => handler(data.params))
+        .then(r => (packet.result = r))
+        .catch(e => (packet.error = e))
+        .then(() => {
+          this.emit('sendReply', packet);
+          this.post(packet);
+        })
+        .catch();
```

Comment:
I'd rm this last catch. No error _should_ happen in the above block, but if one does, we shouldn't squelch it.

Reply:
This is actually picked up by the linter - it throws the "Promises must be properly handled" error without the catch. I think I removed a tslint disable comment. Thoughts?

Reply:
I'd redisable that rule. It's sensible for most code, but for here it's not.